core/resources.rs

```diff
@@ -23,24 +21,27 @@ type ResourceMap = HashMap<ResourceId, Box<dyn Resource>>;
 
 #[derive(Default)]
 pub struct ResourceTable {
-  map: ResourceMap,
+  // TODO(bartlomieju): remove pub modifier, it is used by
+  // `get_file` method in CLI
+  pub map: ResourceMap,
   next_id: u32,
 }
 
 impl ResourceTable {
-  pub fn get<T: Resource>(&self, rid: ResourceId) -> Result<&T, Error> {
-    let resource = self.map.get(&rid).ok_or_else(bad_resource)?;
-    let resource = &resource.downcast_ref::<T>().ok_or_else(bad_resource)?;
-    Ok(resource)
+  pub fn get<T: Resource>(&self, rid: ResourceId) -> Option<&T> {
+    if let Some(resource) = self.map.get(&rid) {
+      return resource.downcast_ref::<T>();
+    }
+
+    None
   }
 
-  pub fn get_mut<T: Resource>(
-    &mut self,
-    rid: ResourceId,
-  ) -> Result<&mut T, Error> {
-    let resource = self.map.get_mut(&rid).ok_or_else(bad_resource)?;
-    let resource = resource.downcast_mut::<T>().ok_or_else(bad_resource)?;
-    Ok(resource)
+  pub fn get_mut<T: Resource>(&mut self, rid: ResourceId) -> Option<&mut T> {
+    if let Some(resource) = self.map.get_mut(&rid) {
```

Comment:
Might as well make get_mut in core return a Option<&mut T> too?

Reply:
Not sure I understand. That's what this function does

Reply:
Err sorry! Misunderstanding on my part.